src/app/ReadClient.cpp

```diff
@@ -1032,11 +1032,7 @@ void ReadClient::OnResubscribeTimerCallback(System::Layer * apSystemLayer, void
         //       defunct has no effect on resident OperationalDeviceProxy instances that are already bound
         //       to a now-defunct CASE session.
         //
-        auto proxy = caseSessionManager->FindExistingSession(_this->mPeer);
-        if (proxy != nullptr)
-        {
-            proxy->Disconnect();
-        }
+        caseSessionManager->DisconnectSession(_this->mPeer);
```

Comment:
We shouldn't need this call at all right? Marking the session as defunct above should be all that is needed.

Reply:
I wasn't sure if the two were guaranteed to be the same peer. Thanks for clearing that up